src/daml/Daml/Trading/Tests/CashRedemptionTest.daml

```diff
@@ -0,0 +1,166 @@
+module Daml.Trading.Tests.CashRedemptionTest where
+
+import Daml.Script
+
+import Daml.Finance.Interface.Types.Common (InstrumentKey(..), AccountKey(..), Id(..))
+import Daml.Trading.Account.Account qualified as CreateAccount
+import Daml.Trading.Cash.CashTransfer qualified as CashTransfer
+import Daml.Trading.Cash.CreditCashAccount qualified as CreditCashAccount
+import Daml.Trading.Scripts.UserSetup qualified as UserSetup (createUsers)
+import Daml.Trading.Scripts.HoldingSetup qualified as HoldingSetup (initialize, instrumentData) 
+import Daml.Trading.Cash.CashRedemption qualified as CashRedemption
+import DA.Optional (fromSome)
+
+setup: Script ()
+setup = do 
```

Comment:
seems like a very long test, is it feasible to split it into several, named scenarios?

Something along these lines:

`shouldSuccessfullyRedeemCash: Script ()
shouldSuccessfullyRedeemCash = do pure ()

shouldFailOnValidationXYZ: Script ()
shouldFailOnValidationXYZ = do pure ()`